src/garage/torch/modules/multi_headed_mlp_module.py

```diff
@@ -61,55 +63,43 @@ def __init__(self,
 
         output_dims = self._check_hidden_layer_parameter(
             'output_dims', output_dims, n_heads)
-        output_nonlinearities = self._check_hidden_layer_parameter(
-            'output_nonlinearities', output_nonlinearities, n_heads)
         output_w_inits = self._check_hidden_layer_parameter(
             'output_w_inits', output_w_inits, n_heads)
         output_b_inits = self._check_hidden_layer_parameter(
             'output_b_inits', output_b_inits, n_heads)
 
-        prev_size = input_dim
-        for size in hidden_sizes:
-            layer = nn.Sequential()
-            if layer_normalization:
-                layer.add_module('layer_normalization',
-                                 nn.LayerNorm(prev_size))
-
-            linear_layer = nn.Linear(prev_size, size)
-            hidden_w_init(linear_layer.weight)
-            hidden_b_init(linear_layer.bias)
-            layer.add_module('linear', linear_layer)
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._layer_normalization = layer_normalization
+        self._output_nonlinearities = self._check_hidden_layer_parameter(
+            'output_nonlinearities', output_nonlinearities, n_heads)
 
-            if hidden_nonlinearity:
-                layer.add_module('non_linearity', hidden_nonlinearity)
+        self._layers = nn.ModuleList()
 
+        prev_size = input_dim
+        for size in hidden_sizes:
+            layer = nn.Linear(prev_size, size)
+            hidden_w_init(layer.weight)
+            hidden_b_init(layer.bias)
             self._layers.append(layer)
             prev_size = size
 
         self._output_layers = nn.ModuleList()
         for i in range(n_heads):
-            layer = nn.Sequential()
-
-            linear_layer = nn.Linear(prev_size, output_dims[i])
-            output_w_inits[i](linear_layer.weight)
-            output_b_inits[i](linear_layer.bias)
-            layer.add_module('linear', linear_layer)
-
-            if output_nonlinearities[i]:
-                layer.add_module('non_linearity', output_nonlinearities[i])
-
+            layer = nn.Linear(prev_size, output_dims[i])
+            output_w_inits[i](layer.weight)
+            output_b_inits[i](layer.bias)
             self._output_layers.append(layer)
 
     def _check_hidden_layer_parameter(self, var_name, var, n_heads):
         if isinstance(var, (list, tuple)):
             if len(var) == 1:
                 var = var * n_heads
             elif len(var) != n_heads:
-                msg = '{} should be either an integer or' \
-                      ' a collection of length n_heads ({}), but {} provided.'
-                raise ValueError(msg.format(var_name, n_heads, var))
+                raise ValueError(f'{var_name} should be matched with number '
```

Comment:
what was wrong with the original message?

Reply:
It wasn't wrong. I just want to make it more clear and use f-strings format in Python3

Reply:
f-strings aren't supported in Python 3.5, so we can't use them in this repo right now (we support Python 3.5+).

Reply:
fixed